core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -52,11 +52,24 @@ static List<Expression> extractWindowExpressions(Iterable<? extends Node> nodes)
                 .build();
     }
 
+    static List<Expression> extractWindowExpressions(Iterable<? extends Node> nodes, Session session, Metadata metadata)
+    {
+        return ImmutableList.<Expression>builder()
+                .addAll(extractWindowFunctions(nodes, session, metadata))
+                .addAll(extractWindowMeasures(nodes))
+                .build();
+    }
+
     static List<FunctionCall> extractWindowFunctions(Iterable<? extends Node> nodes)
     {
         return extractExpressions(nodes, FunctionCall.class, ExpressionTreeUtils::isWindowFunction);
     }
 
+    static List<FunctionCall> extractWindowFunctions(Iterable<? extends Node> nodes, Session session, Metadata metadata)
```

Comment:
If we need to pass the full `Session` and `Metadata` arguments in order to correctly identify window functions in call cases, we should probably remove the existing method `extractWindowFunctions(Iterable<? extends Node>)` to avoid accidentally using it in an inappropriate context, no?

Reply:
I am not entirely sure if we can replace all `extractWindowFunctions` occurrences with the new arguments, I will need to check and ensure that there are no side effects. It makes sense to do so or I can probably rename the function to specify that it is called in the  `verifyNoAggregateWindowOrGroupingFunctions()` stage only.